project/driver_controls/test/test_driver_fsm.c

```diff
@@ -0,0 +1,285 @@
+#include "event_arbiter.h"
+#include "unity.h"
+#include "status.h"
+#include "log.h"
+#include "input_event.h"
+#include "test_helpers.h"
+
+#include "power_fsm.h"
+#include "pedal_fsm.h"
+#include "direction_fsm.h"
+#include "turn_signal_fsm.h"
+#include "hazard_light_fsm.h"
+#include "mechanical_brake_fsm.h"
+
+typedef struct FSMGroup {
+  FSM power;
+  FSM pedal;
+  FSM direction;
+  FSM turn_signal;
+  FSM hazard_light;
+  FSM mechanical_brake;
+} FSMGroup;
+
+static FSMGroup s_fsm_group;
+static bool powered = false;
+
+static void prv_toggle_power(bool new_state) {
+  Event e;
+  if (new_state != powered) {
+    e.id = INPUT_EVENT_POWER;
+    event_arbiter_process_event(&e);
+    powered = new_state;
+  }
+}
+
+void setup_test() { }
+
+void teardown_test(void) {
+  Event e;
+
+  // Release gas pedal
+  e.id = INPUT_EVENT_GAS_BRAKE;
+  event_arbiter_process_event(&e);
+  TEST_ASSERT_EQUAL_STRING("state_brake", s_fsm_group.pedal.current_state->name);
+
+  // Shift gear to neutral
+  e.id = INPUT_EVENT_MECHANICAL_BRAKE;
+  event_arbiter_process_event(&e);
+
+  e.id = INPUT_EVENT_DIRECTION_SELECTOR_NEUTRAL;
+  event_arbiter_process_event(&e);
+  TEST_ASSERT_EQUAL_STRING("state_neutral", s_fsm_group.direction.current_state->name);
+
+  e.id = INPUT_EVENT_MECHANICAL_BRAKE;
+  event_arbiter_process_event(&e);
+
+  prv_toggle_power(false);
+  TEST_ASSERT_EQUAL_STRING("state_off", s_fsm_group.power.current_state->name);
+}
+
+void test_setup() {
+  TEST_ASSERT_OK(event_arbiter_add_fsm(&s_fsm_group.power, power_fsm_init));
+  TEST_ASSERT_OK(event_arbiter_add_fsm(&s_fsm_group.pedal, pedal_fsm_init));
+  TEST_ASSERT_OK(event_arbiter_add_fsm(&s_fsm_group.direction, direction_fsm_init));
+  TEST_ASSERT_OK(event_arbiter_add_fsm(&s_fsm_group.turn_signal, turn_signal_fsm_init));
+  TEST_ASSERT_OK(event_arbiter_add_fsm(&s_fsm_group.hazard_light, hazard_light_fsm_init));
+  TEST_ASSERT_OK(event_arbiter_add_fsm(&s_fsm_group.mechanical_brake, mechanical_brake_fsm_init));
+
+  TEST_ASSERT_EQUAL_STRING("state_off", s_fsm_group.power.current_state->name);
+  TEST_ASSERT_EQUAL_STRING("state_brake", s_fsm_group.pedal.current_state->name);
+  TEST_ASSERT_EQUAL_STRING("state_neutral", s_fsm_group.direction.current_state->name);
+  TEST_ASSERT_EQUAL_STRING("state_no_signal", s_fsm_group.turn_signal.current_state->name);
+  TEST_ASSERT_EQUAL_STRING("state_hazard_off", s_fsm_group.hazard_light.current_state->name);
+  TEST_ASSERT_EQUAL_STRING("state_disengaged", s_fsm_group.mechanical_brake.current_state->name);
+}
+
+void test_power_fsm() {
```

Comment:
The standard for test naming is `test_[module]_[testname]`. So in this case, it would be `test_driver_fsm_[testname]`.

I'd suggest breaking this test into a number of focused tests per-module, maybe with test FSMs, and having a few integration-style tests that make sure all the FSMs work together. The biggest thing with testing is that we want to do some basic sanity checks to make sure it works, but you don't need to go overboard with testing every possible scenario. Unit testing is good to make sure you hit potential edge cases.